src/Compilers/CSharp/Test/Emit2/Semantics/RefReadonlyParameterTests.cs

```diff
@@ -2758,4 +2760,1022 @@ .maxstack  2
+                static void Main()
+                {
+                    var x = 123;
+                    var c = new C();
+                    c.M(ref x);
+                    c.M(in x);
+                    c.M(x);
+                }
+            }
+            """;
+        var verifier = CompileAndVerify(source, expectedOutput: """
+            C.M123
+            C.M123
+            C.M123
+            """, sourceSymbolValidator: verify, symbolValidator: verify);
+        verifier.VerifyDiagnostics(
+            // (7,29): warning CS9507: Reference kind modifier of parameter 'ref readonly int x' doesn't match the corresponding parameter 'in int x' in overridden or implemented member.
+            //     protected override void M(ref readonly int x) => System.Console.WriteLine("C.M" + x);
+            Diagnostic(ErrorCode.WRN_OverridingDifferentRefness, "M").WithArguments("ref readonly int x", "in int x").WithLocation(7, 29),
+            // (14,13): warning CS9503: Argument 1 should be passed with 'ref' or 'in' keyword
+            //         c.M(x);
+            Diagnostic(ErrorCode.WRN_ArgExpectedRefOrIn, "x").WithArguments("1").WithLocation(14, 13));
+
+        static void verify(ModuleSymbol m)
+        {
+            VerifyRequiresLocationAttributeSynthesized(m);
+
+            var p = m.GlobalNamespace.GetMember<MethodSymbol>("C.M").Parameters.Single();
+            VerifyRefReadonlyParameter(p, modreq: true);
+        }
+    }
+
+    [Fact]
+    public void Overridden_In_RefReadonly_Indexer()
+    {
+        var source = """
+            class B
+            {
+                protected virtual int this[in int x]
+                {
+                    get
+                    {
+                        System.Console.WriteLine("B" + x);
+                        return 0;
+                    }
+                    set { }
+                }
+            }
+            class C : B
+            {
+                protected override int this[ref readonly int x]
+                {
+                    get
+                    {
+                        System.Console.WriteLine("C" + x);
+                        return 0;
+                    }
+                    set { }
+                }
+                static void Main()
+                {
+                    var x = 123;
+                    var c = new C();
+                    _ = c[ref x];
+                    _ = c[in x];
+                    _ = c[x];
+                }
+            }
+            """;
+        var verifier = CompileAndVerify(source, expectedOutput: """
+            C123
+            C123
+            C123
+            """, sourceSymbolValidator: verify, symbolValidator: verify);
+        verifier.VerifyDiagnostics(
+            // (22,9): warning CS9507: Reference kind modifier of parameter 'ref readonly int x' doesn't match the corresponding parameter 'in int x' in overridden or implemented member.
+            //         set { }
```

Comment:
I've considered moving this warning from the accessor method to the indexer declaration, but seems other simliar warnings also behave like this, for example nullable type mismatch warnings ([SharpLab](https://sharplab.io/#v2:EYLgZgpghgLgrgJwgZwLQCUJk1AJgewDsAbATwB8ABAJgAIAhAWACgBvF2z2gBwXxggBjAbloA3AJYJ4UYrUoAWWgFkAFPmAArITAD8tAB4BKWgF4AfLRgALPgHdahOMWIBCANwcuvfjoijJaThZWglCGCtrCWQAbQ1tYX0DAF0vTnZmLizaAHMICItaAAZPTOzOAHoK5HzaVloAXzTGlibmGloAYVoQBhYMrJ8BYX9afDEIBAQJXAh5JTV4nUMTQpt7R2c3UsG+YZExiamZubCIm2i4rWWU5oHy3NrCkuasqpqI+rastoagA===)).